### PR TITLE
Initial upload of WIP partitionq module

### PR DESCRIPTION
**Pull request by AIIX**
Add support for WIP partitionq module, All details at https://github.com/calamares/calamares/issues/1954

This request provides the following:
- Initializing the Partition Core Module
- Displaying the Initial install choose page with a working device selector combo box
- Displaying the install choices supported by the device
- Displaying swap choices supported when selecting the "Erase" install option
- Encryption selection and setting the encryption passphrase
- Flat partition list mode derived from partition model  (to be use able with qml list, repeater and combo box components)

### CODE REVIEW

File: src/modules/partitionq/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,107 @@
+# === This file is part of Calamares - <https://calamares.io> ===
+#
+#   SPDX-FileCopyrightText: 2020 Adriaan de Groot <groot@kde.org>
+#   SPDX-License-Identifier: BSD-2-Clause
+#
+if( NOT WITH_QML )
+    calamares_skip_module( "localeq (QML is not supported in this build)" )
+    return()
```
Comment (demmm) [inline src/modules/partitionq/CMakeLists.txt:9]: Seems a copy/paste error, should be partitionq, not localeq, copyright header seems to have wrong name, certainly wrong year

File: src/modules/partitionq/partitionq.conf [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+# value is used instead. It is still a problem if there is no required
+# size set at all, and the replace and resize options will not be offered
+# if no required size is set.
+#
+# The value is in Gibibytes (GiB).
+#
+# BIG FAT WARNING: except for OEM-phase-0 use, you should be using
+#                  the welcome module, **and** configure this value in
+#                  `welcome.conf`, not here.
+# requiredStorage: 3.5
```
Comment (demmm) [inline src/modules/partitionq/partitionq.conf]: Usually the .conf for a QML references the corresponding widgets .conf, only adds QML specific settings, not duplicate all instructions

Comment (AIIX) [reply src/modules/partitionq/partitionq.conf]: Fixed in latest commit, have removed all the partitionq.conf file settings, but left the file in there with a to do message, as I am not sure yet what settings actually need to be in QML module itself, will remove it if no settings live in qml in the future

File: src/modules/partitionq/CMakeLists.txt (hunk 1 above)
Comment (AIIX) [reply src/modules/partitionq/CMakeLists.txt:9]: Fixed in latest commit

**Pull request merged**